bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/RuleRegistry.java

```diff
@@ -115,4 +129,70 @@
      */
     public void runNow(String ruleUID);
 
+    /**
+     * Creates a {@link Predicate} which can be used to filter {@link Rule}s for a given scope or {@code null} scope.
+     * @param scope to search for
+     * @return created {@link Predicate}
+     */
+    public static Predicate<Rule> hasScope(String scope) {
+        if (null == scope) {
+            return r -> null == r.getScope();
+        } else {
+            return r -> scope.equals(r.getScope());
+        }
+    }
+
+    /**
+     * Creates a {@link Predicate} which can be used to filter {@link Rule}s for any of the given scopes and even {@code null} scope.
+     * @param scopes to search for
+     * @return created {@link Predicate}
+     */
+    public static Predicate<Rule> hasAnyScope(String... scopes) {
+        HashSet<String> scopeSet = new HashSet<String>(scopes.length);
+        for(final String scope : scopes) {
+            scopeSet.add(scope);
+        }
+
+        // this will even work for null scopes
+        return r-> scopeSet.contains(r.getScope());
+    }
+
+    /**
+     * Creates a {@link Predicate} which can be used to filter {@link Rule}s for a all given tags or {@link Rule}s without tags.
+     * All given tags must match, (the matched {@code Rule} might contain more).
+     * @param tags to search for
+     * @return created {@link Predicate}
+     */
+    public static Predicate<Rule> hasAllTags(Collection<String> tags) {
+        if (null == tags || tags.isEmpty()) {
```

Comment:
The difference is that `Predicate`s can be combined and are way more powerful e.g. the current API can't be easily negated.

Example:
`getAll(hasAllTags(someTags).negate().or(hasScope(myScope)))`
Currently it's not possible to get the same result set using the current APi of `getBy...()`.
The Predicate is easy to write and one does not have to get all `Rule`s using `getAll()` and then do all the filtering manually.